pkg/checks/checks.go

```diff
@@ -0,0 +1,210 @@
+package checks
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appclient "k8s.io/client-go/kubernetes/typed/apps/v1"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// WaitForHTTPStatusOk poll until URL returns 200
+func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urltocheck string) error {
+	cli := &http.Client{
+		Transport: transport,
+		Timeout:   10 * time.Second,
+	}
+
+	req, err := http.NewRequest("GET", urltocheck, nil)
+	if err != nil {
+		return err
+	}
+
+	for {
+		resp, err := cli.Do(req)
+		if err, ok := err.(*url.Error); ok && (err.Timeout() || err.Err == io.EOF || err.Err == io.ErrUnexpectedEOF) {
+			time.Sleep(time.Second)
+			continue
+		}
+
+		if err == io.EOF {
+			time.Sleep(time.Second)
+			continue
+		}
+		if err != nil {
+			log.Printf("the err was %#v\n", err)
+			return err
+		}
+		if resp.StatusCode == http.StatusOK {
+			return nil
+		}
+
+		select {
+		case <-ctx.Done():
+			log.Printf("ctx.Done called.  Returning err")
+			return ctx.Err()
+		case <-time.After(time.Second):
+		}
+	}
+}
+
+// WaitForInfaServices verify daemonsets, statefulsets
+func WaitForInfaServices(ctx context.Context, appclient *appclient.AppsV1Client) error {
```

Comment:
@jim-minter was this more in-line with what you had in mind?

Should I add the deeper dive (fetch pods and check status) when I find `ReadyReplicas != Replicas` with each type?

Reply:
yes :)